main.js

```diff
@@ -0,0 +1,114 @@
+let BOOKS_DATA = [];
+
+function setStorage() {
+  localStorage.setItem('books_data', JSON.stringify(BOOKS_DATA));
+}
+
+function retrieveStorage() {
+  BOOKS_DATA = JSON.parse(localStorage.getItem('books_data'));
+}
+
+if (!localStorage.getItem('books_data')) {
+  setStorage();
+} else {
+  retrieveStorage();
+}
+
+let form = document.querySelector('form');
+let bookList = document.getElementById('book-list');
+
+function updateSelections() {
+  form = document.querySelector('form');
+  bookList = document.getElementById('book-list');
+}
+
+function addBook() {
+  const bookTitle = document.getElementById('book-title');
+  const bookAuthor = document.getElementById('book-author');
+
+  const bookObj = {};
+
+  let exists = false;
+
+  for (let i = 0; i < BOOKS_DATA.length; i += 1) {
+    if (BOOKS_DATA[i].title === bookTitle.value && BOOKS_DATA[i].author === bookAuthor.value) {
+      exists = true;
+    }
+  }
+
+  if (!exists) {
+    bookObj.title = bookTitle.value;
+    bookObj.author = bookAuthor.value;
+    return bookObj;
+  }
+
+  return null;
+}
+
+function removeBook(element) {
+  const textElements = element.parentElement.querySelectorAll('.book-card-text');
+
+  BOOKS_DATA = BOOKS_DATA.filter((obj) => {
+    if (textElements[0].innerText === obj.title && textElements[1].innerText === obj.author) {
+      return false;
+    }
+
+    return true;
+  });
+}
+
+function generateBookCard(bookObj) {
+  const bookCard = document.createElement('li');
+  const bookTitle = document.createElement('p');
+  const bookAuthor = document.createElement('p');
+  const removeBtn = document.createElement('button');
+
+  bookCard.classList.add('book-card', 'mb-1');
+  bookTitle.classList.add('book-card-text');
+  bookAuthor.classList.add('book-card-text');
+  removeBtn.classList.add('remove-book', 'mb-05');
+
+  bookTitle.textContent = bookObj.title;
+  bookAuthor.textContent = bookObj.author;
```

Comment:
- [ ] [OPTIONAL] This project is a small one and some good practices might not seem important. But it would help us to practice some concepts like Object Destructuring. The variables `title` and `author` can be extracted at the top of `generateBookCard` method and used inside the function body:

```JavaScript
const { title, author } = bookObj;
```

After that, do away with the dot notation:

```JavaScript
bookTitle.textContent = title;
bookAuthor.textContent = author;
```